spec/inputs/rabbitmq_spec.rb

```diff
@@ -54,15 +60,42 @@
       subject { hare_info }
 
       it "should set the queue correctly" do
+        register_instance
         expect(subject.queue).to eql(queue)
       end
 
       it "should set the prefetch value correctly" do
+        register_instance
         expect(channel).to have_received(:prefetch=).with(123)
       end
 
       context "with an exchange declared" do
-        let(:instance) { rabbitmq_settings.merge("exchange" => "myexchange", "exchange_type" => "fanout") }
+        @manual_register = true
+        let(:exchange) { double("exchange") }
+        let(:key) { double("routing key") }
+        let(:rabbitmq_settings) { super.merge("exchange" => exchange, "key" => key) }
+
+        it "should bind to the exchange" do
+          register_instance
+          expect(queue).to have_received(:bind).with(exchange, :routing_key => key)
+        end
+
+        context "but not immediately available" do
+          before do
+            setup_mocks
```

Comment:
Given this pattern here, I think the top-level `before do ...` should call setup_mocks, then you can do per-context `before do` ... to call register or do any other setup.